Tasks/Common/azure-arm-rest/azure-arm-endpoint.ts

```diff
@@ -43,15 +43,17 @@ export class AzureRMEndpoint {
 
 
             this.endpoint.authenticationType =  tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'authenticationType', true);
-            if(this.endpoint.authenticationType && this.endpoint.authenticationType == constants.AzureServicePrinicipalAuthentications.servicePrincipalCertificate) {
-                tl.debug('certificate spn endpoint');
-                this.endpoint.servicePrincipalCertificate = tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'servicePrincipalCertificate', false);
-                this.endpoint.servicePrincipalCertificatePath = certFilePath;
-                fs.writeFileSync(this.endpoint.servicePrincipalCertificatePath, this.endpoint.servicePrincipalCertificate);
-            }
-            else {
-                tl.debug('credentials spn endpoint');
-                this.endpoint.servicePrincipalKey = tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalkey', false);
+            if (this.endpoint.scheme && this.endpoint.scheme.toLowerCase() === constants.AzureRmEndpointAuthenticationScheme.ServicePrincipal) {
```

Comment:
Initial we had settled on if scheme is null or service principal then parse ServicePrincipalKey ? Are we planning to revert this changes ?